### PR TITLE
Added full support for TELNET proxy

### PR DESCRIPTION
**Pull request by cstayyab**
Added a struct for telnet Proxy and added an instance of it in the Gutton struct. Also registered it as a handler in glutton.go
Added an example for the rule of telnet_proxy in examples folder. 
In protocol .go the telnet handler is called same as other handlers are called.

### CODE REVIEW

File: proxy_telnet.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+	t := &telnetProxy{
+		logger:  g.logger,
+		glutton: g,
+	}
+
+	dest, err := url.Parse(destinationURL)
+	if err != nil {
+		t.logger.Error("[telnet.prxy] failed to parse destination address, check config.yaml")
+		fmt.Println(fmt.Sprintf("[telnet.prxy ]   failed to parse destination address, check config.yaml"))
```
Comment (glaslos) [inline proxy_telnet.go]: remove this print statement

File: proxy_telnet.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+		return err
+	}
+	t.logger.Info("[telnet proxy] " + dest.Host)
```
Comment (glaslos) [inline proxy_telnet.go]: use fmt.Sprintf here

File: proxy_telnet.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+	return
+}
+
+//handle Telnet Proxy handler
+func (t *telnetProxy) handle(ctx context.Context, conn net.Conn) (err error) {
+	ended := false
+	g := t.glutton
+	tcpAddr, err := net.ResolveTCPAddr("tcp", t.host)
+	if err != nil {
+		println("ResolveTCPAddr failed:", err.Error())
```
Comment (glaslos) [inline proxy_telnet.go]: use t.logger.Error

File: proxy_telnet.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		return
+	}
+	hconn, err := net.DialTCP("tcp", nil, tcpAddr)
+	if err != nil {
+		t.logger.Error(fmt.Sprintf("[telnet proxy  ]  Connection error: %v", err))
+		fmt.Println(fmt.Sprintf("[telnet proxy ]  Coonection error: %v", err))
```
Comment (glaslos) [inline proxy_telnet.go]: remove print

File: proxy_telnet.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+		for {
+			if ended == true || hconn == nil || conn == nil {
+				break
+			}
+			reply := make([]byte, 8*1024)
+			n, err = hconn.Read(reply)
+			if err != nil {
+				if err == io.EOF {
+					t.logger.Error(fmt.Sprintf("[telnet proxy  ]   Connection closed by Server"))
+					fmt.Println(fmt.Sprintf("[telnet proxy ]  Connection closed by Server"))
```
Comment (glaslos) [inline proxy_telnet.go]: remove print

File: proxy_telnet.go [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+					break
+				}
+				t.logger.Error(fmt.Sprintf("[telnet proxy  ]   error: %v", err))
+				fmt.Println(fmt.Sprintf("[telnet proxy ]  error: %v", err))
```
Comment (glaslos) [inline proxy_telnet.go]: remove print

File: proxy_telnet.go [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+				break
+			}
+
+			t.logger.Info(fmt.Sprintf("[telnet proxy  ]   Info: Recieved: %d bytes(s) from Server. Bytes: %s", len(string(reply)), string(reply)))
+			fmt.Println(fmt.Sprintf("[telnet proxy ]  Info: Recieved: %d bytes(s) from Server", len(string(reply))))
```
Comment (glaslos) [inline proxy_telnet.go]: remove print

File: proxy_telnet.go [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+			writeMsg(conn, string(reply), g)
```
Comment (glaslos) [inline proxy_telnet.go]: this returns an error

File: proxy_telnet.go [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+			conn.Close()
+			hconn.Close()
+		}()
+		for {
+			if ended == true || hconn == nil || conn == nil {
+				return
+			}
+			msg, err := readMsg(conn, g)
+			if err != nil {
+				break
```
Comment (glaslos) [inline proxy_telnet.go]: log error here?

File: proxy_telnet.go [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+			}
+			_, err = hconn.Write([]byte(msg))
+			if err != nil {
+				if err == io.EOF {
+					t.logger.Error(fmt.Sprintf("[telnet proxy  ]   Connection closed by Server"))
+					fmt.Println(fmt.Sprintf("[telnet proxy ]  Connection closed by Server"))
```
Comment (glaslos) [inline proxy_telnet.go]: remove print

File: proxy_telnet.go [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+		defer func() {
+			ended = true
+			conn.Close()
+			hconn.Close()
+		}()
+		for {
+			if ended == true || hconn == nil || conn == nil {
+				break
+			}
+			reply := make([]byte, 8*1024)
```
Comment (glaslos) [inline proxy_telnet.go]: what if more is send?

Comment (cstayyab) [reply proxy_telnet.go]: Please notice line no. 59 in which I have called this function as a go routine. That means that this function is running asynchronously. I will continue to fetch the data from connection and continuously output to screen. So it it does not matter whatever value I put here or whatever the size of array is.

I hope I have made myself clear.

**Pull request merged**